### PR TITLE
add option for zlib compression in .nc file exports

### PR DESCRIPTION
**Pull request by fneum**
closes #614 

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] Unit tests for new features were added (if applicable).
- ~[ ] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).~
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [x] I consent to the release of this PR's code under the MIT license.

### CODE REVIEW

File: doc/release_notes.rst [hunk 1]
```diff
@@ -21,6 +21,9 @@ Upcoming Release
 
 * Fix a bug where time-dependant generator variables could be forgotten during aggregation in a particular case.
 
+* New argument for file compression in ``n.export_to_netcdf(path,  compress=True)``
```
Comment (fneum) [inline doc/release_notes.rst:24]:
```suggestion
* New argument for file compression in ``n.export_to_netcdf(path,  compression=True)``
```

File: doc/release_notes.rst [hunk 2]
```diff
@@ -21,6 +21,9 @@ Upcoming Release
 
 * Fix a bug where time-dependant generator variables could be forgotten during aggregation in a particular case.
 
+* New argument for file compression in ``n.export_to_netcdf(path,  compression=True)``
+  will typecast network datasets to float32 and use ``zlib`` for compression.
```
Comment (fneum) [inline doc/release_notes.rst:25]:
```suggestion
* New argument for file compression in ``n.export_to_netcdf(path,  compression=True)``
  will typecast network datasets to float32 and use ``zlib`` for compression. The option `least_significant_digit` was removed.
```

**Review by FabianHofmann — APPROVED** [review]

**Comment by fneum** [comment]
Do you understand why the tests fail?

**Comment by FabianHofmann** [comment]
Honestly, no idea... perhaps new version of netcdf? Or it is the CI servers....

**Comment by fneum** [comment]
Indeed, some problems with libnetcdf >= 4.9.

**Comment by fneum** [comment]
Windows / Python 3.10 test failure is unrelated.

**Pull request merged**